### PR TITLE
[compile](fix) fix gcc compile error

### PR DESCRIPTION
**Pull request by shuke987**
## Proposed changes
fcc reports error: comparing the result of pointer addition when compile
`
if (places[i] + offset)
`

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by shuke987** [comment]
run buildall

**Comment by shuke987** [comment]
run buildall

**Comment by shuke987** [comment]
run performance

File: be/src/vec/aggregate_functions/aggregate_function_collect.h [hunk 1]
```diff
@@ -678,7 +678,7 @@ class AggregateFunctionCollect
                                             Arena* arena, const size_t num_rows) const override {
         if constexpr (ShowNull::value) {
             for (size_t i = 0; i != num_rows; ++i) {
-                if (places[i] + offset) {
+                if (places[i] || offset) {
```
Comment (cambyzju) [inline be/src/vec/aggregate_functions/aggregate_function_collect.h:681]: We should not change the logic while fix compile error. 

offset is added from pr https://github.com/apache/doris/pull/32387, please have a review @amorynan

Comment (shuke987) [reply be/src/vec/aggregate_functions/aggregate_function_collect.h:681]: place[i] is a pointer, and offset is an unsigned int, so if placces[i] != null or offset != 0, equals places[i] + offset != null

Comment (cambyzju) [reply be/src/vec/aggregate_functions/aggregate_function_collect.h:681]: maybe the correct answer is like: `if(places[i] && offset < places[i].size())`, check `offset != 0 or == 0` is strange here.

Comment (zclllyybb) [reply be/src/vec/aggregate_functions/aggregate_function_collect.h:681]: here below we use `places[i]+offset` as an pointer. so maybe `places[i] + offset != nullptr` is correct and more readable

**Pull request closed without merging**